### PR TITLE
More ergonomic profile name handling

### LINKED ISSUE #6156: [CT-1426] Easier programmatic profile selection
Per a discussion with the switchboard team this PR allows for overriding the profile name.  It represents a minor improvement  of the existing code and will be oblivated by the Click profile/config work, but there's an urgent need for this now ergo this ticket/PR

### PR DESCRIPTION
**Pull request by iknox-fa**
resolves #6156

### Description
Allows for overriding the profile name when creating new runtime configs.  
N.B. Per SB team we need to backport this to 1.2.

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] I have [opened an issue to add/update docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose), or docs changes are not required/relevant for this PR
- [x] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

File: core/dbt/config/runtime.py [hunk 1]
```diff
@@ -3,31 +3,42 @@
 from copy import deepcopy
 from dataclasses import dataclass, field
 from pathlib import Path
-from typing import Dict, Any, Optional, Mapping, Iterator, Iterable, Tuple, List, MutableSet, Type
```
Comment (iknox-fa) [inline core/dbt/config/runtime.py:6]: FYI-- Most of these import changes are just sorting... Nothing major changed here but having properly sorted imports matters so I'm doing a little cleanup whilst here.

File: core/dbt/config/runtime.py [hunk 2]
```diff
@@ -190,28 +201,52 @@ def _get_rendered_profile(
 
     @classmethod
     def collect_parts(cls: Type["RuntimeConfig"], args: Any) -> Tuple[Project, Profile]:
```
Comment (iknox-fa) [inline core/dbt/config/runtime.py:203]: The code in this function will be basically what the SB team will use to generate the profile and project, which can then be used with the `from_parts` method to generate Runtime configs.

The difference will be potentially overriding the profile name when it's collected like so:
```
profile = cls.collect_profile(args=args, profile_name="a_valid_profile_name")
```

**Comment by leahwicz** [comment]
@iknox-fa will this need to go into 1.3 as well then? Do we see any risk or breaking change with this going into a patch? (still catching up on this in Slack...)

**Review by ChenyuLInx — APPROVED** [review]
LGTM

**Comment by ChenyuLInx** [comment]
> @iknox-fa will this need to go into 1.3 as well then? Do we see any risk or breaking change with this going into a patch? (still catching up on this in Slack...)

@leahwicz  this won't be a breaking change.

**Comment by iknox-fa** [comment]
> @iknox-fa will this need to go into 1.3 as well then? Do we see any risk or breaking change with this going into a patch? (still catching up on this in Slack...)

Yep-- apparently they support 1.2+ so we'll ned this change in both.  Do I need anotherPR for that?  (I forget how our backport system works.. I haven't had to use it in ages).

**Pull request merged**